File: feature-translate/src/main/java/com/andyanika/translator/feature/translate/TranslationView.java
package com.andyanika.translator.feature.translate;


import com.andyanika.translator.common.models.ui.DisplayTranslateResult;

public interface TranslationView {
    void showTranslation(DisplayTranslateResult result);
    void showProgress();
    void hideProgress();
    void showErrorLayout();
    void hideErrorLayout();
    void showClearBtn();
    void hideClearBtn();
    void clearResult();
    void clearTranslation();
    void setSrcLabel(String text);
    void setDstLabel(String text);
    void showOffline();
    void hideOffline();
}

File: feature-translate/src/main/java/com/andyanika/translator/feature/translate/TranslationView.kt
package com.andyanika.translator.feature.translate

import com.andyanika.translator.common.models.ui.DisplayTranslateResult

interface TranslationView {
    fun showTranslation(result: DisplayTranslateResult)
    fun showNotFound()
    fun showProgress()
    fun hideProgress()
    fun showErrorLayout()
    fun hideErrorLayout()
    fun showClearBtn()
    fun hideClearBtn()
    fun clearResult()
    fun clearTranslation()
    fun setSrcLabel(text: String?)
    fun setDstLabel(text: String?)
    fun showOffline()
    fun hideOffline()
}
